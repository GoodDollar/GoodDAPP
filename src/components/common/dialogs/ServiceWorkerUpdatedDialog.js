// libraries
import React from 'react'
import { Platform, View } from 'react-native'

// components
import { t } from '@lingui/macro'
import Text from '../../common/view/Text'
import JumpingPeopleAnimation from '../animations/JumpingPeople'

// utils
import { withStyles } from '../../../lib/styles'
import { getDesignRelativeHeight, getDesignRelativeWidth, isSmallDevice } from '../../../lib/utils/sizes'
import { isBrowser } from '../../../lib/utils/platform'

// assets
import UpdateVersionSVG from '../../../assets/updateversion.svg'

// localization

const dialogStyles = ({ theme }) => {
  const topPosition = isBrowser ? 0 : isSmallDevice ? 40 : 80

  const animation = {
    height: getDesignRelativeHeight(144),
    width: getDesignRelativeHeight(279),
    marginBottom: getDesignRelativeHeight(45),
    position: 'relative',
    top: getDesignRelativeHeight(topPosition),
  }

  return {
    animation,
    image: {
      textAlign: 'center',
      height: getDesignRelativeHeight(134),
      width: getDesignRelativeHeight(191),
      marginBottom: getDesignRelativeHeight(25),
      justifyContent: 'center',
    },
    imageContainer: {
      paddingTop: getDesignRelativeHeight(32),
      display: 'flex',
      alignItems: 'center',
    },
    phase1Title: {
      ...Platform.select({
        web: { borderTopStyle: 'solid' },
      }),
      borderTopWidth: 2,
      borderTopColor: theme.colors.primary,
      paddingTop: getDesignRelativeWidth(9),
    },
    title: {
      width: '100%',
      paddingBottom: getDesignRelativeWidth(9),
      marginBottom: getDesignRelativeWidth(9),
      borderBottomWidth: 2,
      borderBottomColor: theme.colors.primary,
      ...Platform.select({
        web: { borderBottomStyle: 'solid' },
      }),
    },
    description: {
      minHeight: getDesignRelativeWidth(100),
    },
  }
}

export const RegularDialog = withStyles(dialogStyles)(({ styles, theme }) => (
  <View>
    <View style={styles.imageContainer}>
      <View style={styles.image}>
        <UpdateVersionSVG />
      </View>
      <Text
        fontSize={22}
        lineHeight={26}
        textAlign="left"
        fontWeight="bold"
        color={theme.colors.darkGray}
        style={styles.title}
      >
        {t`Hip hip hooray!
        We have a new version :)`}
      </Text>
    </View>
    <View style={styles.description}>
      <Text fontSize={14} lineHeight={20} textAlign="left" color={theme.colors.darkGray}>
        {t`Thanks to your awesome feedback we were able to improve our wallet. We hope now you’ll enjoy it even more.`}
      </Text>
    </View>
  </View>
))

export const NewReleaseDialog = withStyles(dialogStyles)(({ styles, theme }) => {
  const textStyles = { fontSize: 14, lineHeight: isSmallDevice ? 30 : 26 }

  return (
    <View>
      <View style={styles.imageContainer}>
        <View style={styles.animation}>
          <JumpingPeopleAnimation />
        </View>
        <Text
          fontSize={26}
          lineHeight={38}
          textAlign="left"
          fontWeight="bold"
          color={theme.colors.green}
          style={[styles.title, styles.phase1Title]}
        >
          {t`Good News: We’re Live!`}
        </Text>
      </View>
      <View style={styles.description}>
        <Text {...textStyles} textAlign="left" color={theme.colors.darkGray}>
          <Text {...textStyles} fontWeight="bold">
            1.
          </Text>
          {t` Click ‘update’`}
          <Text {...textStyles} fontWeight="bold">
            2.
          </Text>
          {t` Sign up (one last time, we promise ${isSmallDevice ? '' : ':)'})`}
          <Text {...textStyles} fontWeight="bold">
            3.
          </Text>
          {t` Start claiming your free, REAL G$’s`}
        </Text>
      </View>
    </View>
  )
})
