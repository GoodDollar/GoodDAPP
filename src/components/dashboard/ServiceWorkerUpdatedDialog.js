import React from 'react'
import { View } from 'react-native'
import { withStyles } from '../../lib/styles'
import Text from '../common/view/Text'
import { getDesignRelativeHeight, getDesignRelativeWidth } from '../../lib/utils/sizes'
import UpdateVersionSVG from '../../assets/updateversion.svg'

const dialogStyles = ({ theme }) => ({
  image: {
    alignItems: 'center',
    height: getDesignRelativeHeight(134),
    width: getDesignRelativeHeight(191),
    marginBottom: getDesignRelativeHeight(25),
  },
  imageContainer: {
    paddingTop: getDesignRelativeHeight(32),
    paddingHorizontal: getDesignRelativeWidth(5),
    display: 'flex',
    alignItems: 'center',
  },
  title: {
    width: '100%',
    borderStyle: 'solid',
    borderBottomWidth: 2,
    borderBottomColor: theme.colors.primary,
    paddingBottom: getDesignRelativeWidth(9),
    marginBottom: getDesignRelativeWidth(9),
  },
  description: {
    minHeight: getDesignRelativeWidth(100),
  },
})

export default withStyles(dialogStyles)(({ styles, theme }) => {
  return (
    <View>
      <View style={styles.imageContainer}>
        <View style={styles.image}>
          <UpdateVersionSVG />
        </View>
        <Text
          fontSize={22}
          lineHeight={26}
          textAlign="left"
          fontWeight="bold"
          color={theme.colors.darkGray}
          style={styles.title}
        >
          {`Hip hip hooray!\nWe have a new version :)`}
        </Text>
      </View>
      <View style={styles.description}>
        <Text fontSize={14} lineHeight={20} textAlign="left" color={theme.colors.darkGray}>
          Thanks to your awesome feedback we were able to improve our wallet. We hope now youâ€™ll enjoy it even more.
        </Text>
      </View>
    </View>
  )
})
