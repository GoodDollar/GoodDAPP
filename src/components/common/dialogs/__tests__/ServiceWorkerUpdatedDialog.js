import React from 'react'

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer'
import { forIn } from 'lodash'

import { NewReleaseDialog, RegularDialog } from '../ServiceWorkerUpdatedDialog'
import { withThemeAndLocalizationProvider } from '../../../../__tests__/__util__'

describe('ServiceWorkerUpdatedDialog', () => {
  forIn({ RegularDialog, NewReleaseDialog }, (component, name) =>
    describe(name, () => {
      const WrappedComponent = withThemeAndLocalizationProvider(component)

      it(`renders without errors`, () => {
        let tree

        expect(() => (tree = renderer.create(<WrappedComponent />))).not.toThrow()
        expect(tree.toJSON()).toBeTruthy()
      })

      it(`matches snapshot`, () => {
        const tree = renderer.create(<WrappedComponent />)

        expect(tree.toJSON()).toMatchSnapshot()
      })
    }),
  )
})
