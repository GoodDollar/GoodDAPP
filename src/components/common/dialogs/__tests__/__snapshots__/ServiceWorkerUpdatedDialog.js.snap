// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ServiceWorkerUpdatedDialog NewReleaseDialog matches snapshot 1`] = `
<div
  className="css-view-1dbjc4n r-flex-13awgt0 r-pointerEvents-12vffkv"
>
  <div
    className="css-view-1dbjc4n"
  >
    <div
      className="css-view-1dbjc4n r-alignItems-1awozwy r-display-6koalj r-paddingTop-wk8lta"
    >
      <div
        className="css-view-1dbjc4n r-height-1w2pmg r-marginBottom-p1pxzi r-position-bnwqim r-top-ipm5af r-width-1h2t8mc"
      >
        <div
          className="css-view-1dbjc4n r-width-13qz1uu"
          style={
            Object {
              "WebkitBoxPack": "end",
              "WebkitJustifyContent": "flex-end",
              "height": "100px",
              "justifyContent": "flex-end",
              "marginTop": "0px",
              "msFlexPack": "end",
            }
          }
        >
          <div
            className="css-view-1dbjc4n"
            style={
              Object {
                "marginBottom": "0px",
                "width": "100%",
              }
            }
          />
        </div>
      </div>
      <div
        className="css-text-901oao r-borderBottomColor-lpwvss r-borderBottomStyle-rull8r r-borderBottomWidth-wgabs5 r-borderTopColor-dw99bl r-borderTopStyle-1efd50x r-borderTopWidth-1jxfwug r-color-qjtdzx r-fontFamily-vnw8o6 r-fontSize-hiw28u r-fontWeight-b88u0q r-lineHeight-qvk6io r-marginBottom-p1pxzi r-paddingBottom-1mdbw0j r-paddingTop-wk8lta r-textAlign-fdjqy7 r-textDecorationLine-13wfysu r-textTransform-3twk1y r-width-13qz1uu"
        dir="auto"
      >
        Good News: We’re Live!
      </div>
    </div>
    <div
      className="css-view-1dbjc4n r-minHeight-ifefl9"
    >
      <div
        className="css-text-901oao r-color-cddv6f r-fontFamily-vnw8o6 r-fontSize-1b43r93 r-fontWeight-1it3c9n r-lineHeight-qvk6io r-textAlign-fdjqy7 r-textDecorationLine-13wfysu r-textTransform-3twk1y"
        dir="auto"
      >
        <span
          className="css-text-901oao css-textHasAncestor-16my406 r-color-cddv6f r-fontFamily-vnw8o6 r-fontSize-1b43r93 r-fontWeight-b88u0q r-lineHeight-qvk6io r-textAlign-q4m81j r-textDecorationLine-13wfysu r-textTransform-3twk1y"
        >
          1.
        </span>
        Click ‘update’
        <span
          className="css-text-901oao css-textHasAncestor-16my406 r-color-cddv6f r-fontFamily-vnw8o6 r-fontSize-1b43r93 r-fontWeight-b88u0q r-lineHeight-qvk6io r-textAlign-q4m81j r-textDecorationLine-13wfysu r-textTransform-3twk1y"
        >
          2.
        </span>
        Sign up (one last time, we promise )
        <span
          className="css-text-901oao css-textHasAncestor-16my406 r-color-cddv6f r-fontFamily-vnw8o6 r-fontSize-1b43r93 r-fontWeight-b88u0q r-lineHeight-qvk6io r-textAlign-q4m81j r-textDecorationLine-13wfysu r-textTransform-3twk1y"
        >
          3.
        </span>
        Start claiming your free, REAL G$’s
      </div>
    </div>
  </div>
</div>
`;

exports[`ServiceWorkerUpdatedDialog RegularDialog matches snapshot 1`] = `
<div
  className="css-view-1dbjc4n r-flex-13awgt0 r-pointerEvents-12vffkv"
>
  <div
    className="css-view-1dbjc4n"
  >
    <div
      className="css-view-1dbjc4n r-alignItems-1awozwy r-display-6koalj r-paddingTop-wk8lta"
    >
      <div
        className="css-view-1dbjc4n r-height-1w2pmg r-justifyContent-1777fci r-marginBottom-p1pxzi r-textAlign-q4m81j r-width-1h2t8mc"
      >
        <updateversion.svg />
      </div>
      <div
        className="css-text-901oao r-borderBottomColor-lpwvss r-borderBottomStyle-rull8r r-borderBottomWidth-wgabs5 r-color-cddv6f r-fontFamily-vnw8o6 r-fontSize-hiw28u r-fontWeight-b88u0q r-lineHeight-qvk6io r-marginBottom-p1pxzi r-paddingBottom-1mdbw0j r-textAlign-fdjqy7 r-textDecorationLine-13wfysu r-textTransform-3twk1y r-width-13qz1uu"
        dir="auto"
      >
        Hip hip hooray!
We have a new version :)
      </div>
    </div>
    <div
      className="css-view-1dbjc4n r-minHeight-ifefl9"
    >
      <div
        className="css-text-901oao r-color-cddv6f r-fontFamily-vnw8o6 r-fontSize-1b43r93 r-fontWeight-1it3c9n r-lineHeight-qvk6io r-textAlign-fdjqy7 r-textDecorationLine-13wfysu r-textTransform-3twk1y"
        dir="auto"
      >
        Thanks to your awesome feedback we were able to improve our wallet. We hope now you’ll enjoy it even more.
      </div>
    </div>
  </div>
</div>
`;
